src/protocol/boolean/bitwise_lt.rs

```diff
@@ -0,0 +1,257 @@
+        b: &[Replicated<F>],
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+    ) -> Result<Vec<Replicated<F>>, BoxError> {
+        let mul = zip(repeat(ctx), zip(g, b))
+            .enumerate()
+            .map(|(i, (ctx, (&g_bit, &b_bit)))| {
+                let c = ctx.narrow(&BitOpStep::Step(i));
+                async move { c.multiply(record_id, g_bit, b_bit).await }
+            });
+        try_join_all(mul).await
+    }
+
+    /// Step 6. `[h] = Σ [h_i] where i=0..l-1`
+    ///
+    /// The interpretations is, `h` is 1 iff `a < b`
+    fn step6<F: Field>(h: &[Replicated<F>]) -> Replicated<F> {
+        h.iter()
+            .fold(Replicated::new(F::ZERO, F::ZERO), |acc, &x| acc + x)
+    }
+
+    #[allow(dead_code)]
+    #[allow(clippy::many_single_char_names)]
+    pub async fn execute<F: Field>(
+        ctx: ProtocolContext<'_, Replicated<F>, F>,
+        record_id: RecordId,
+        a: &[Replicated<F>],
+        b: &[Replicated<F>],
+    ) -> Result<Replicated<F>, BoxError> {
+        debug_assert_eq!(a.len(), b.len(), "Length of the input bits must be equal");
+        let mut e = Self::step1(a, b, ctx.narrow(&Step::AXorB), record_id).await?;
+        let f = Self::step2(&mut e, ctx.narrow(&Step::PrefixOr), record_id).await?;
+        let g = Self::step3_4(&f);
+        let h = Self::step5(&g, b, ctx.narrow(&Step::MaskLessThanBit), record_id).await?;
+        let result = Self::step6(&h);
+        Ok(result)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+enum Step {
+    AXorB,
+    PrefixOr,
+    MaskLessThanBit,
+}
+
+impl crate::protocol::Substep for Step {}
+
+impl AsRef<str> for Step {
+    fn as_ref(&self) -> &str {
+        match self {
+            Self::AXorB => "a_xor_b",
+            Self::PrefixOr => "prefix_or",
+            Self::MaskLessThanBit => "mask_less_than_bit",
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::BitwiseLessThan;
+    use crate::{
+        error::BoxError,
+        ff::{Field, Fp31},
+        protocol::{QueryId, RecordId},
+        secret_sharing::Replicated,
+        test_fixture::{make_contexts, make_world, share, validate_and_reconstruct, TestWorld},
+    };
+    use futures::future::try_join_all;
+    use rand::rngs::mock::StepRng;
+    use std::iter::zip;
+
+    type BitwiseShares = (
+        Vec<Replicated<Fp31>>,
+        (Vec<Replicated<Fp31>>, Vec<Replicated<Fp31>>),
+    );
+
+    fn fp31_to_bits(x: Fp31) -> Vec<Fp31> {
+        let x = u8::from(x);
+        (0..u8::BITS).map(|i| Fp31::from(x >> i & 1)).collect()
+    }
+
+    async fn bitwise_lt(a: Fp31, b: Fp31) -> Result<Fp31, BoxError> {
+        let world: TestWorld = make_world(QueryId);
+        let ctx = make_contexts::<Fp31>(&world);
+        let mut rand = StepRng::new(1, 1);
+
+        let a_bits = fp31_to_bits(a);
+        let b_bits = fp31_to_bits(b);
+
+        // Generate secret shares
+        #[allow(clippy::type_complexity)]
```

Comment:
ngl, clippy ain't wrong about this...

I get that you want helper-wise vectors of stuff, and not a single vector of triples of stuff... but this is kind of gnarly.

You could produce a simpler data structure, (e.g. make the `fp31_to_bits()` function just return `Vec<(Replicated<F>, Replicated<F>, Replicated<F>)>`) and then when you want to invoke the BitwiseLessThan::execute() function map this into what it is expecting, i.e.
```
BitwiseLessThan::execute(..., a_bits.map(|bit| bit.0), b_bits.map(|bit| bit.0)),
...
```